varats/varats/gui/cs_gen/main_window.py

```diff
@@ -0,0 +1,353 @@
+"""Main Window of the case study generation Gui."""
+import sys
+import typing as tp
+from datetime import datetime, timezone, timedelta
+from enum import Enum
+from pathlib import Path
+
+import benchbuild as bb
+import pygit2
+from PyQt5.QtCore import (
+    QModelIndex,
+    QDateTime,
+    Qt,
+    QSortFilterProxyModel,
+    QAbstractTableModel,
+)
+from PyQt5.QtGui import QColor
+from PyQt5.QtWidgets import QMainWindow, QApplication, QMessageBox
+
+from varats.base.sampling_method import NormalSamplingMethod
+from varats.gui.cs_gen.main_window_ui import Ui_MainWindow
+from varats.mapping.commit_map import (
+    create_lazy_commit_map_loader,
+    get_commit_map,
+    CommitMap,
+)
+from varats.paper.case_study import CaseStudy, store_case_study
+from varats.paper_mgmt.case_study import (
+    extend_with_extra_revs,
+    extend_with_distrib_sampling,
+    extend_with_revs_per_year,
+)
+from varats.project.project_util import (
+    get_loaded_vara_projects,
+    get_local_project_git_path,
+    get_local_project_git,
+    get_project_cls_by_name,
+)
+from varats.projects.discover_projects import initialize_projects
+from varats.revision.revisions import is_revision_blocked
+from varats.tools.research_tools.vara_manager import ProcessManager
+from varats.ts_utils.cli_util import initialize_cli_tool
+from varats.utils import settings
+from varats.utils.git_util import (
+    get_initial_commit,
+    get_all_revisions_between,
+    ShortCommitHash,
+    create_commit_lookup_helper,
+    FullCommitHash,
+)
+from varats.utils.settings import vara_cfg
+
+
+class GenerationStrategie(Enum):
+    """Enum for the Startegie used when Generating a CaseStudy."""
+    SELECTREVISION = 0
```

Comment:
```suggestion
    SELECT_REVISION = 0
```
To follow the naming schema, this needs an `_` or?